tests/e2e/registry/test_registry_reboot_node.py

```diff
@@ -0,0 +1,195 @@
+        nodes.restart_nodes(node)
+
+        # Validate all nodes and services are in READY state and up
+        retry(
+            (CommandFailed, TimeoutError, AssertionError, ResourceWrongStatusException),
+            tries=60,
+            delay=15)(
+            wait_for_cluster_connectivity(tries=400)
+        )
+        retry(
+            (CommandFailed, TimeoutError, AssertionError, ResourceWrongStatusException),
+            tries=60,
+            delay=15)(
+            wait_for_nodes_status(timeout=900)
+        )
+
+        # Validate image registry pods
+        validate_registry_pod_status()
+
+        # Validate image exists in registries path
+        validate_image_exists(namespace=self.project_name)
+
+        # Validate cluster health ok and all pods are running
+        self.sanity_helpers.health_check()
+
+    @pytest.mark.parametrize(
+        argnames=['node_type'],
+        argvalues=[
+            pytest.param(
+                *[MASTER_MACHINE], marks=pytest.mark.polarion_id("OCS-1802")
+            ),
+            pytest.param(
+                *[WORKER_MACHINE], marks=pytest.mark.polarion_id("OCS-1804")
+            ),
+        ]
+    )
+    def test_registry_rolling_reboot_node(self, node_type, nodes):
+        """
+        Test registry workload when backed by OCS and reboot node one by one
+        """
+
+        # Get the node list
+        node_list = get_typed_nodes(node_type)
+
+        # Pull and push images to registries
+        log.info("Pull and push images to registries")
+        image_pull_and_push(
+            project_name=self.project_name, template='eap-cd-basic-s2i',
+            image='registry.redhat.io/jboss-eap-7-tech-preview/eap-cd-openshift-rhel8:latest',
+            pattern='eap-app'
+        )
+
+        # Validate image exists in registries path
+        validate_image_exists(namespace=self.project_name)
+
+        for node in node_list:
+
+            # Reboot node
+            log.info(node)
+            nodes.restart_nodes([node], wait=False)
+
+            # Wait some time after rebooting node
+            waiting_time = 40
+            log.info(f"Waiting {waiting_time} seconds...")
+            time.sleep(waiting_time)
+
+            # Validate all nodes and services are in READY state and up
+            retry(
+                (CommandFailed, TimeoutError, AssertionError, ResourceWrongStatusException),
+                tries=60,
+                delay=15)(
+                wait_for_cluster_connectivity(tries=400)
+            )
+            retry(
+                (CommandFailed, TimeoutError, AssertionError, ResourceWrongStatusException),
+                tries=60,
+                delay=15)(
+                wait_for_nodes_status(timeout=900)
+            )
+
+            # Validate image registry pods
+            validate_registry_pod_status()
+
+            # Validate image exists in registries path
+            validate_image_exists(namespace=self.project_name)
+
```

Comment:
Also validate that all the storage pods are in running state post node recovery.

Reply:
below function sanity_helpers.health_check(), validates all nodes are in Ready state, all storage pods are in running and also ceph health is OK. Do we need to add another check for only storage pods ?